test/Microsoft.Azure.WebJobs.Host.UnitTests/Singleton/SingletonEnd2EndTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using System;
+using System.Threading;
+using Xunit;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Host.UnitTests.Singleton
+{
+    public class SingletonEnd2EndTests 
```

Comment:
But that uses external storage and is in the FunctionTests project. This has no dependency so can rightfully live in the UnitTest project.  but has the added benefit that it exercises the entire end-2-end [Singleton] machinery. 

---
In reply to: [120952102](https://github.com/Azure/azure-webjobs-sdk/pull/1184#discussion_r120952102) [](ancestors = 120952102)